### ISSUE TITLE
Transfer Learning: good results for RPN, but bad for final result

### ISSUE THREAD

**Issue by ninafiona** [matterport/Mask_RCNN#2589]
I have my own dataset with 1 class (+ background) with 100 images in train and 16 in validation dataset. I trained (based on coco weights) 8 epochs (1000 steps) on heads, 3 epochs for 4+ and 4 epochs for all layers. I can see on tensorboard that the validation loss went down like exp(-x) - as expected. 

Now I looked at my model using inspect_model.ipynb and unfortunately my results don't look good at all. I did see though that the RPN predictions looked perfect, for N=5 my object was always detected by all bounding boxes. 

Do I have to train for more epochs? Any other hints on why it didn't work well? 

I wasn't sure how to set my steps, I have 2 images per gpu (=batch size) and 100 training images, does it make sense to set steps to 50 then? I tried it with 50 steps and epochs 10 (stage 1), 5 (stage 2) and 55 (stage 3), but same outcome as above.

**Comment by ninafiona** [comment]
I found my error. I loaded the weights like this:
`model.load_weights(weights_path, by_name=True, exclude=[ "mrcnn_class_logits", "mrcnn_bbox_fc", "mrcnn_bbox", "mrcnn_mask"])`

changing to 
`model.load_weights(model_path, by_name=True)`
resolved my error. 

See https://github.com/matterport/Mask_RCNN/issues/1209
